### PR TITLE
ABR 10a: Unit tests for RangeFetcher classes

### PR DESCRIPTION
**Pull request by gsheasby**
**Goals (and why)**: Add unit test coverage before ABR rollout

**Implementation Description (bullets)**:
- Added TokenRangeFetcherTest
- Added RepairRangeFetcherTest
- Actually fixed a couple of small bugs along the way (specifically, making the token ranges more closely match those given by Cassandra, including the "wraparound range". 

**Testing (What was existing testing like?  What have you done to improve it?)**:
Added a bunch of unit tests 🚀 

These tests are very similar to those in CassandraRepairEteTest - my main reason for not removing them yet is that the ones in CRET also cover some type-massaging code from CassandraRepairHelper - I'll probably cover that code and remove the ETE tests in a later PR.

**Concerns (what feedback would you like?)**: Any test cases missing? (check TokenRangeFetcher and RepairRangeFetcher)

**Where should we start reviewing?**: TokenRangeFetcherTest and RepairRangeFetcherTest most likely.

**Priority (whenever / two weeks / yesterday)**: would like this to merge this week, but after ABR9.

### CODE REVIEW

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/backup/CassandraRepairHelper.java
Comment (gsheasby) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/backup/CassandraRepairHelper.java:90]: This is fine for now - context:

I looked into the internal code and chatted with Leon about this - the parallelism logic used internally is quite complex, and we agree that for the purposes of this workstream, it should be fine to keep the parallel side of things in IBP ( (internal backup product).
This means that IBP would call the AtlasBackupService/AtlasRestoreService with one namespace at a time, but would do these calls in parallel.
In practice, the tasks for a single namespace are not done in parallel (for a single namespace, they repair each table in serial, then clean each table in serial) - so it would not harm perf to have IBP call one Atlas restore method that does the repair + clean in serial, provided IBP called Atlas with one namespace at a time. This also makes wiring the Atlas code up to IBP more straightforward.
If we want to further improve performance in future, we could look at moving the parallelism into Atlas and making it more granular, but that’s not needed to satisfy this workstream’s objective, and would add both overhead and risk to do it properly - thus, it’s out of scope.

I’d be happy to add this to an ADR once the IBP wiring is in place, so that we have a more persistent place that we can refer to later, if someone decides to build on it as a hack week project or something

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/backup/CqlMetadata.java
Comment (gmaretic) [inline atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/backup/CqlMetadata.java:81]: I think the answer is no, but just to double-check: this doesn't do any kind of DNS loookup?

**Review by gmaretic — APPROVED** [review]
One question, but otherwise 🚢

File: atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/backup/CqlMetadata.java
Comment (gsheasby) [reply atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/backup/CqlMetadata.java:81]: No - the EndPoint implementation used is a TranslatedAddressEndPoint, which just wraps the InetSocketAddress that we gave it when configuring the cluster.

**Comment by gsheasby** [comment]
Adding autorelease so that I can start working AtlasBackupService and AtlasRestoreService into IBP, but that stuff won't merge until #5827 is merged.

**Pull request merged**
